explanation.md

```diff
@@ -0,0 +1,38 @@
+#Explanation of spy function
+
+##Basic structure
+
+The basic structure of the file is as follows:
+
+```
+var firstAnonFunc = function(global, factory){}
+var secondAnonFunc = function(fn, ctx){ return spyfn }
+firstAnonFunc(this, secondAnonFunc);
+```
+Using the (function(){}()) notation, the first anonymous function is executed as it is defined. This function is passed the global context held in the `this` variable and another anonymous function which in turn will generate the spy function whenever it is called.
+
+##The first anonymous function
+The role of the first anonymous function is to make the second anonymous function callable from any point in our script. This is done in one of three ways depending on the version of JavaScript. The simplest way is to simply assign the second anonymous function to a global variable named `spyOn`.
```

Comment:
Is not about the version of JavaScript, is about the way of load javascript files. It could be `commonJS` (this is how node require other files), `Asynchronous Module Definition`  (is how RequireJS usually loads files) or as a global variable.